certora/specs/RecoveryConfirmationSignatureValidity.spec

```diff
@@ -0,0 +1,131 @@
+using SafeHarness as safeContract;
+
+methods {
+    // Social Recovery Module Functions
+    function nonce(address) external returns (uint256) envfree;
+    function isGuardian(address, address) external returns (bool) envfree;
+
+    // Social Recovery Module Summaries
+    function getRecoveryHash(address, address[] calldata, uint256, uint256) internal returns (bytes32) => CONSTANT;
+    function SignatureChecker.isValidSignatureNow(address signer, bytes32 dataHash, bytes memory signatures) internal returns (bool) => isValidSignatureNowSummary(signer, dataHash, signatures);
+
+    // Wildcard Functions
+    function _.execTransactionFromModule(address to, uint256 value, bytes data, Enum.Operation operation) external with (env e) => summarizeSafeExecTransactionFromModule(calledContract, e, to, value, data, operation) expect bool ALL;
+    function _.isModuleEnabled(address module) external => DISPATCHER(false);
+    function _.isOwner(address owner) external => DISPATCHER(false);
+    function _.getOwners() external => DISPATCHER(false);
+    function _._ external => DISPATCH[] default NONDET;
+}
+
+
+// A summary function that helps the prover resolve calls to `safeContract`.
```

Comment:
I don't know if this is needed, actually, I copied it from the original `SocialRecoveryModule.spec`

Reply:
Maybe @remedcu has more context about this? I remember Certora mentioning that we can use `DISPATCHER(false)` as a shorthand for these kinds of summaries.

Reply:
Turns out, this one is a bit of a mystery. We should have used a list in the dispatcher instead of `false` in this case, but as this results in timeout, we tried with `false`. Even that results in error for some reason and doesn't have a problem when we summarize it.

Reply:
What do you mean by "a list in the dispatcher"?

The [docs](https://docs.certora.com/en/latest/docs/cvl/methods.html#dispatcher-summaries) mention that DISPATCHER only accepts `true` or `false`

Reply:
https://docs.certora.com/en/latest/docs/cvl/methods.html#catch-unresolved-calls-entry

My bad, not dispatcher, but `DISPATCH`.

> Can we add a brief description to the summary in the "main" spec file with this so we don't lose this knowledge (separate PR is fine)?

I am checking with the Certora team if this can be solved with just `DISPATCHER(false)` like others. If not, we can make a PR with the text:

```
Here we are using summary as summarization never reverts. If you call a solidity function from the summary, the prover assumes it never reverts and thus will not exhibit any reverting behaviour. If we use a `DISPATCHER(false)` then there is a possibility of ghost states havoc-ing resulting in errors.
```